src/proto/h2/server.rs

```diff
@@ -331,18 +362,22 @@ enum H2StreamState<F, B>
 where
     B: HttpBody,
 {
-    Service(#[pin] F),
+    Service(#[pin] F, Option<(Pending, Recorder, RecvStream)>),
```

Comment:
What do you think of turning this into a struct variant? Mostly just to give a label to what this `Option` is for?

Reply:
Oh, does that play well with pin projection? Maybe it's an annoying change, and just a doc comment would be enough to remind the next spelunker.

Reply:
I think pin projection handles struct variants just fine, I'll try to name the fields.

Reply:
Rather than naming the fields of the enum variant, I introduced a new structure for the triplet.